pandas/core/arrays/string_arrow.py

```diff
@@ -363,48 +364,139 @@ def __setitem__(self, key: int | slice | np.ndarray, value: Any) -> None:
         """
         key = check_array_indexer(self, key)
 
-        if is_integer(key):
-            key = cast(int, key)
+        if is_list_like(key):
+            key = np.asarray(key)
+            if len(key) == 1:
+                key = key[0]
 
-            if not is_scalar(value):
-                raise ValueError("Must pass scalars with scalar indexer")
-            elif isna(value):
+        value_is_scalar = is_scalar(value)
+
+        # NA -> None
```

Comment:
i would create a helper method like `_validate_key()` to encapsulate all of this (ok on this class for now, but we likey want to push this to the ArrowExtensionArray (or maybe we need a ArrowIndexingMixin or similar), that can be later (or here if convenient).

Reply:
I refactored pretty extensively, this logic is now self contained